pkg/controller/druid/handler.go

```diff
@@ -151,6 +154,18 @@ func deployDruidCluster(sdk client.Client, m *v1alpha1.Druid) error {
 				return err
 			}
 		}
+
+		// Create HPA Spec
+		if nodeSpec.AutoScale != nil {
+			if err := sdkCreateOrUpdateAsNeeded(sdk,
+				func() (object, error) {
+					return makeHorizontalPodAutoscaler(&nodeSpec, m, ls, nodeSpecUniqueStr)
+				},
+				func() object { return makeHorizontalPodAutoscalerEmptyObj() },
+				nil, m, hpaName); err != nil {
+				return err
+			}
+		}
```

Comment:
can you also add section like `updatedStatus.StatefulSets = deleteUnusedResources(sdk, m, statefulSetNames, ls,...)`  for autoscaler resource so that it gets GC'd when not needed and also correctly reflected in status, See later for similar snippets for other resources we have?